module/data/actor/fields/travel-field.mjs

```diff
@@ -0,0 +1,148 @@
+import MappingField from "../../fields/mapping-field.mjs";
+
+const { StringField } = foundry.data.fields;
+
+/**
+ * @typedef {"slow"|"normal"|"fast"} TravelPace5e
+ */
+
+/**
+ * @typedef {object} TravelData
+ * @property {TravelPace5e} [pace]            Current travel pace.
+ * @property {Record<string, string>} paces   Formulas for various travel paces per/day.
+ * @property {Record<string, string>} speeds  Formulas for various travel speeds per/hour.
+ * @property {string} units                   Movement used to measure the various travel speeds.
+ */
+
+/**
+ * Field for storing travel data.
+ */
+export default class TravelField extends foundry.data.fields.SchemaField {
+  constructor(fields={}, { initialUnits=null, ...options }={}) {
+    fields = {
+      pace: new StringField({
+        required: true, blank: false, initial: "normal", choices: () => CONFIG.DND5E.travelPace,
+        label: "DND5E.Travel.Label"
+      }),
+      paces: new MappingField(new FormulaField({ deterministic: true }), {
+        initialKeys: CONFIG.DND5E.travelTypes, initialKeysOnly: true
+      }),
+      speeds: new MappingField(new FormulaField({ deterministic: true }), {
+        initialKeys: CONFIG.DND5E.travelTypes, initialKeysOnly: true
+      }),
+      units: new StringField({
+        required: true, nullable: true, blank: false, initial: initialUnits, label: "DND5E.UNITS.TRAVEL.Label"
+      }),
+      ...fields
+    };
+    Object.entries(fields).forEach(([k, v]) => !v ? delete fields[k] : null);
+    super(fields, options);
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Number of hours of travel per day.
+   * @type {number}
+   */
+  static #HOURS_PER_DAY = 8; // TODO: Allow this to be configured
+
+  /* -------------------------------------------- */
+  /*  Data Preparation                            */
+  /* -------------------------------------------- */
+
+  /**
+   * Prepare travel data.
+   * @this {ActorDataModel}
+   * @param {object} rollData  Actor's roll data.
+   */
+  static prepareData(rollData) {
+    const { movement, travel } = this.attributes;
+    const { pace: paceMode, units } = travel;
+    Object.defineProperty(travel.paces, "max", { value: 0, writable: true });
+    Object.defineProperty(travel.speeds, "max", { value: 0, writable: true });
+
+    const noMovement = this.parent.hasConditionEffect("noMovement");
+    const halfMovement = this.parent.hasConditionEffect("halfMovement");
+    const multiplier = noMovement ? 0 : halfMovement ? 0.5 : 1;
+    for ( const type of Object.keys(travel.speeds) ) {
+      let pace = travel.paces[type] = Math.max(0, simplifyBonus(travel.paces[type], rollData)) * multiplier;
+      const speed = travel.speeds[type] = Math.max(0, simplifyBonus(travel.speeds[type], rollData)) * multiplier;
+      if ( pace > travel.paces.max ) travel.paces.max = pace;
+      if ( speed > travel.speeds.max ) travel.speeds.max = speed;
```

Comment:
I think we need to check these after we've done the derivations below.

Reply:
Because we use these for displaying on the sheet, I decided to stick them before the derivation so that only speeds you explicitly enter are displayed on the sheet to give creators a bit more control over the presentation.